ComSemApp/templates/ComSemApp/student/assessment.html

```diff
@@ -154,24 +168,37 @@ <h2 id="timer-text" class="show-expression"></h2>
   attemptFormData.append('student', student);
   attemptFormData.append('audio', audioBlob);
   var csrf_token = '{{ csrf_token }}'; // grabs csrf_token for certification
-  $.ajax({ // send POST request
-        type: "POST",
-        url: $("#save_attempt_url").val(), // links to html form above which specifies the url to recieve request
-        data: attemptFormData,
-        processData: false,
-        contentType: false,
-        headers: { // places token in request so that it is certified by server
-           'X-CSRFToken': csrf_token
-        },
-        success: function(response){
-            cs_notification('success', "Attempt Saved")
-        },
-        error: function(jqXHR, textStatus, errorThrown){
-            cs_ajax_error(jqXHR, textStatus, errorThrown)
-        }
-  });
 
-  initializeRecorder();
+  // returns a promise that calls the ajax, if successful resolves, if error OR takes more than 5 seconds
+  // to return then rejects
+  return new Promise((resolve, reject)=>{
+    $.ajax({ // send POST request
+      type: "POST",
+      url: $("#save_attempt_url").val(), // links to html form above which specifies the url to recieve request
+      data: attemptFormData,
+      processData: false,
+      contentType: false,
+      headers: { // places token in request so that it is certified by server
+        'X-CSRFToken': csrf_token
+      },
+      success:(response) => {
+        cs_notification('success', "Attempt Saved")
+        // inputs the id recieved into the html form 
+        let id = response['id'];
+        let line = "<input type='hidden' value="+id+" name='attempt_ids'></input>";
+        $("#form").append(line);
+        resolve();
+      },
+      error: (jqXHR, textStatus, errorThrown) => {
+        cs_ajax_error(jqXHR, textStatus, errorThrown);
+        reject();
+      }
+    });
+    setTimeout(()=>{
+      console.log("timeout here");
+      reject();
```

Comment:
It might be useful to issue a cs_notification to let the user know it's timed out

Reply:
i decided to change this all together and use the ajax param "timeout" with 5000 ms as it works a bit better. It returns to the error field of the ajax if 5 secs goes by and it gives the text status "timeout" which is then thrown as a cs notification error